climpred_skeleton/core.py

```diff
@@ -0,0 +1,28 @@
+import xarray as xr
+
+
+class Verification:
+    """Absolute base class. Begins the pipeline of verifying a forecast.
+
+    Holds a prediction that is being verified against observations. Stores
+    helpful base attributes (`nmember`, etc.) and runs checks on input
+    data.
+    """
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
+        # Check that inputs are xarray objects, convert to dataset.
+        # Convert to `cftime`.
+        self._prediction = prediction
+        self._observation = observation
+        if 'member' in self._prediction.dims:
+            self._nmember = self._prediction['member'].size
+            self._members = self._prediction['member'].data
+        else:
+            self._nmember, self._members = None, None
+        self._all_verifs = self._observation['time'].data
+        self._all_inits = self._prediction['init'].data
+
+    def _drop_members(self, members: list = None):
```

Comment:
why is this function needed here?

Reply:
Since this is the base class, maybe the Urclass, everything beneath it inherits all of its methods. So we can define convenience methods here that will be useful to all other objects below. We use `_drop_members` in the `m2c` comparison and might need it elsewhere. So now whenever you need to run drop members on a later object/class, you just run `._drop_members` (see in my `m2c` implementation)

Reply:
I think we will never need a drop_member call anywhere. but good, I understood the concept here.